[quick-start]simple-solution-for-node-setup (#31)
bitmark-inc/docs: learning-bitmark/quick-start/simple-solution-for-node-setup.md

#  [Learning Bitmark / Quick start] - Simple solution for node setup

### Start your bitmark-node

> This section will quide you to start a bitmark docker node. The **docker** node, also called **bitmark-node**, is a bitmark full-node. A full-node can generate blocks, mine the blocks and do verification of payments. 

Reviewer: Does `bitmark-node` only refer to the node running inside a docker container?

Reviewer 2: Do you think we should introduce a bit about the bitmark node in general first, then back to the content of the section? 
Just for reference, here is the text from the technical writer for the section `Run a node` which will be removed from the website:

>Unlike conventional property systems that rely on centralized gatekeepers, the Bitmark blockchain is an open and transparent property system that is strengthened through the active participation of anyone on the Internet. You can support the integrity of Bitmark’s open-source blockchain by running the Bitmark node software and becoming a member of its peer-to-peer network. You will earn money and property rewards by doing so.

After that introduction, we could mention this section, such as:
>There are different ways to run a Bitmark node, in this section, we introduce a simple way to do so with running a node contributed as a docker container image. ...

Author: Is this ok , anything to add !? 
Start your bitmark node
A user can run a bitmark node in several ways. This section will guide you to start a bitmark node using docker container. Inside a bitmark node docker container, there are bitmarkd, recorderd, payment system and a user interface to control and monitor services.

### Assumptions
+ [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) is installed in your system
+ [docker](https://docs.docker.com/install/) is installed in your system
    +  Windows user run docker on Hyper-V
+ you [log in](https://docs.docker.com/engine/reference/commandline/login/) your docker with [docker hub]((https://hub.docker.com/)) account
+ you have and know your **public IP** address
+ If you are behind a firewall, you know how to setup a **port-forwarding**
+ you have **a bitcoin and a litecoin address** for recieving mining reward

## Installation

### Step 0: Preparation
+ Have your **Public IP** ready

> + Setup port-forwarding if nessasary

Reviewer: Readers might not remember what makes it necessary to set up port forwarding?

Reviewer 2: 1. I think the list under `Assumptions` is more about `Prerequites`. 
2. Why do we need `git` installed?

Author: @nodestory as discussion change to
```+ You may need to setup your network, ie. port forwarding, in order to allow other bitmark nodes to communicate with your bitmark node.```
@vienlt  
1. I can change to "Prerequisites" . Libra uses Assumptions. They may have a subtle difference but I think we don't need to argue and maybe leave it to Technical writer. 
2. Remove git

Author: @vienlt  You also may consider We have a preparation step so is a prerequisite a better word?  I'll let you decide

    + execute bash install-node-linux-mac.sh [Your Public IP]. When run the script without providing [Your Public IP], the script will find your IP from Internet

        ```bash install-node-linux-mac.sh 123.123.123.123```
    + follow script instruction and select "1)Installation"

    
+  Windows users
    +  ensure that Hyper-V is turned on
    +  [download script for windows](https://s3-ap-northeast-1.amazonaws.com/bitmark-node-docker-scripts/bitmarkNode-HyperV.bat)
    + execute bitmarkNode-HyperV.bat [Your Public IP]. When run the script without providing [Your Public IP], the script find your IP from Internet

        ``` bitmarkNode-HyperV.bat 117.166.111.123```
    + follow script instruction and select "1)Installation"
+ execute **docker ps** in your command-line/shell to check if bitmark-node is running
![docker ps result](https://i.imgur.com/l3dF4Hl.jpg)

## Management Panel

### Step 1: Open A Node-Management Webpage 
+ view our node-mangement webpage on http://127.0.0.1:9980 in a browser
+ input your BTC and LTC address for rewarding in a pop-up screen
![Input your BTC&LTC Address](https://i.imgur.com/IRTlyjY.jpg?1)

> ### Step 2: Run bitmarkd node and recorderd

Reviewer: ```suggestion
### Step 2: Run bitmarkd and recorderd
```
bitmarkd node = `bitmarkd` + `recorderd`?

Reviewer 2: Can we explain what the script does?

+ Startup screen
    + Run bitmarkd by click "start button" of "Bitmark Node (bitmarkd)"
    + After bitmarkd is running, run a recorderd by click "start button" of "Recorder Node (recorderd)"
![startup screen](https://i.imgur.com/aeONALb.jpg)

> + Running screen

Reviewer: Shall we teach them how to identify if the node is successfully or not?

Reviewer 2: How can users access the screen?

Author: @nodestory As our discussion, I will add a check section in the bottom. 
@vienlt ,  in step 1
Step 1: Open A Node-Management Webpage
view our node-mangement webpage on http://127.0.0.1:9980 in a browser

#  [ Learning Bitmark / Quick start ] - Simple solution for node setup

## Start your bitmark node

A user can run a bitmark node in several ways. This section will guide you to start a bitmark node using docker container. Inside a bitmark node docker container, there are bitmarkd, recorderd, payment system and a user interface to control and monitor services.

> 

Reviewer 3: If you're going to mention multiple ways, it's probably worth mentioning what the other methods are. It's definitely worth noting this is the simplest, quickest, and most reliable setp method.

Reviewer 3: And I see I missed your question here, about which is best: "option 1 A user can run a bitmark node in several ways.
option 2 You can run a bitmark node in several ways.
@nodestory: I personally would like to avoid both options. My option is
There are several ways to run a bitmark node."

The last one is passive voice, where you don't have an active subject. I would usually avoid it unless there are no options (and sometimes there aren't).

The other two are both fine. Saying "A user" is more formal and "you" is more casual. I personally like "you" because it's very direct. The biggest thing is just to be consistent. If you regularly address the readers as "a user" (or other noun), don't use "you", and if you regularly use "you", use it whenever you're directly addressing the readers. So, make a style decision, and we can just ensure that it's used.

## Prerequisites

+ [Docker](https://docs.docker.com/install/) is installed in your system

>     +  Windows user run docker on Hyper-V

Reviewer 3: This seems confusing: readers won't necessarily understand it, and it's already covered in the Docker docs (in fact, it looks like the Docker install automatically turns on Hyper-V)

Author: @shannona  some widows do not support hyper-v ie, home-edition but I can not figure out them all. It depends on users to find out themselves. What is your suggestion for this situation?

+ You [log in](https://docs.docker.com/engine/reference/commandline/login/) your docker with [docker hub]((https://hub.docker.com/)) account
+ You have and know your **public IP** address
+ You have **a bitcoin and a litecoin address** for receiving mining reward
+ Your environment is open for incoming connections through port 2136.  

## Installation

### Step 1: Download and Run Script

Downloaded script will help you to manage bitmark node container. The  installation of the script will pull down a latest bitmark node container image from docker hub and then start a bitmark node container for you.  To check what functions the script supports and how to run it, simply run ``` bash install-node-linux-mac.sh```.  or ```bitmarkNode-HyperV.bat```.

> 

Reviewer 3: You should say which system each of these commands runs on:
(on Mac and Linux)
(on Windows)

Author: i see

+  Linux and mac users
    + [Download scripts for linux and mac](https://bitmark-node-docker-scripts.s3-ap-northeast-1.amazonaws.com/install-node-linux-mac.sh)

>     + Execute bash install-node-linux-mac.sh [Your Public IP]. When run the script without providing [Your Public IP], the script will find your IP from Internet

Reviewer 3: In-line commands should usually be marked for clarity: `bash install-node-linux-mac.sh [Your Public IP]`

Reviewer 3: If you're just going to list it in a full example (as you do here), then you can probably have something shorter in the description, e.g.:

* Execute the install script using `bash`, including your Public IP address. If you do not know your IP address, you can omit that, and the script will try to determine it from the internet:

        ```bash install-node-linux-mac.sh  <Your Public IP>```

Author: Can you give me an example of shortening the description?  It is very short for my point of view. What do you mean in-line command should be marked for clarity?   I have a ``` ``` (Comment on that command and also ident it)

        ```bash install-node-linux-mac.sh  <Your Public IP>```
    + Follow script instruction and select "1)Installation"

    
+  Windows users
    +  Ensure that Hyper-V is turned on
    +  [Download script for windows](https://s3-ap-northeast-1.amazonaws.com/bitmark-node-docker-scripts/bitmarkNode-HyperV.bat)
    + Execute bitmarkNode-HyperV.bat [Your Public IP]. When run the script without providing [Your Public IP], the script find your IP from Internet

        ``` bitmarkNode-HyperV.bat 117.166.111.123```

>     + Follow script instruction and select "1)Installation"

Reviewer 3: Some comment on the style of how you depict the example command.

+ Execute **docker ps** in your command-line/shell to check if bitmark-node is running

> ![docker ps result](https://i.imgur.com/l3dF4Hl.jpg)

Reviewer 3: And here, you probably want `docker ps` not **docker ps** for consistency. 

I think the style we've developed elsewhere is `command` or **App Menu**

Author: do you mean ```docker ps``` not  ***docker ps*** in markdown ? 
"I think the style we've developed elsewhere is command or App Menu" ? Do you mean we have a guideline or example?

    + Run bitmarkd by click "start button" of "Bitmark Node (bitmarkd)"
    + After bitmarkd is running, run a recorderd by click "start button" of "Recorder Node (recorderd)"
    +  If you don't want to use your machine to help us on mining, you can stop recorderd.

![startup screen](https://i.imgur.com/aeONALb.jpg)

+ Running screen
    + Wait for 5 connections to establish a distribute network
    + The "Current Block" shows the progress of blocks synchronization of the node
    + "Your Blocks" shows the blocks that this account has won

![running screen](https://i.imgur.com/g9baqm8.jpg)

If block does not start to  synchronize  with remote bitmark node over 20 minutes, check your network. Also see [Important Notice On Network
](#important-notice-on-network ) 

## Important Notice On Network

User must ensure the following ports to be accessible from Internet.

| PORT | DESCRIPTION                                     |
|------|--------------------------------------------------|
| 2136 | Port for connecting to other peer bitmarkd nod      |

When running bitmark node container, user can make sure ports are opened with the following commands.

netcat -v [Your Public IP] 2136

> 

Reviewer 3: Does that work on Windows, Linux, and Mac?

Author: Basically, the answer is no for windows but Windows support WSL, I have not tested it.  Only test it on ubuntu/Linux. We assume that users know how to debug their net and we provide them the direction.

Author: find the way to test on windows

WebUI is an interface to control bitmark node services. User can only access WebUI through local network. Please notice that user can not access the port from internet due to security reason.

> 

Reviewer 3: This feels a little out of place here currently. I think you need to do one of two things:

1) Have a summary at the start of this section that notes that Bitmark uses two ports, 2136 and 9980, but that they serve different purposes and are accessed differently; 

2) or move the 9980 discussion to where it's most appropriate, which is probably "Step 1" above.

#  [ Learning Bitmark / Quick start ] - Simple solution for node setup

## Start your bitmark node

A user can run a bitmark node in several ways. This section will guide you to start a bitmark node using docker container. Inside a bitmark node docker container, there are bitmarkd, recorderd, payment system and a user interface to control and monitor services.

## Prerequisites

+ [docker](https://docs.docker.com/install/) is installed in your system
    +  Windows user run docker on Hyper-V
+ you [log in](https://docs.docker.com/engine/reference/commandline/login/) your docker with [docker hub]((https://hub.docker.com/)) account
+ you have and know your **public IP** address
+ you have **a bitcoin and a litecoin address** for receiving mining reward

> + you may need to setup your network to allow other bitmark nodes to communicate with your bitmark node. You can check more information on "Important Notice On Network" below.

Reviewer 2: Please capitalize the first character of a bullet except for the special cases. In this section, I think the `you` should be `You`.

Reviewer 2: I couldn't find the `"Important Notice On Network"`, could you link the text to the corresponding section?

Reviewer: - your environment is open for incoming connections through port 2136

And I see we try to guide readers to `Important Network Notes` more than once, and I guess readers are very likely to be forced to scroll down to the bottom for that section. Does it make sense that we won't guide the readers to `Important Network Notes` section until they find the node is not running properly?

And are there other common problems when running a node? Could `Important Network Notes` be a more general troubleshooting section?

Author: changed as request

## Installation

### Step 0: Preparation
+ Have your **Public IP** ready
+ Make sure following ports are accessible from internet
    + port	**2136** for connecting to other peer bitmarkd nodes
    + port	**2135** for publishing blockchain events

>     + port 	**2130** for Bitmark node RPC server

Reviewer: Maybe we don't require ports 2130 and 2135 to be publicly accessible. Users can use localhost to subscribe events and send rpc commands with their own node only.

Author: ok

You can check more information on "Important Notice On Network" below.

> 

Reviewer 2: `port` to `Port` for each bullet.

Reviewer 2: `from internet` to `from the Internet`.

Reviewer 2: >I couldn't find the "Important Notice On Network", could you link the text to the corresponding section? And do you mean Important Network Notes?

Author: have add link

### Step 1: Download and Run Script

Downloaded script will help you to manage bitmark node container. The  installation of the script will pull down a latest bitmark node container image from docker hub and then start a bitmark node container for you.  To check what functions the script supports and how to run it, simply run ``` bash install-node-linux-mac.sh```.  or ```bitmarkNode-HyperV.bat```.

> 

Reviewer 2: Suggestions:

- `Downloaded script` to `The script`

- `will pull down a latest...` to `will pull down the latest...`

Personally I think the last sentence is a bit confusing because the `bash` command is to run the script instead of checking how to run it.

+  Linux and mac users
    + [download scripts for linux and mac](https://bitmark-node-docker-scripts.s3-ap-northeast-1.amazonaws.com/install-node-linux-mac.sh)
    + execute bash install-node-linux-mac.sh [Your Public IP]. When run the script without providing [Your Public IP], the script will find your IP from Internet

>         ```bash install-node-linux-mac.sh 123.123.123.123```

Reviewer 2: A suggestion about the format:

- Run the downloaded script:
   ```shell
      $ bash install-node-linux-mac.sh <Your Public IP>
   ````
    > **Note:** In case the Public IP isn't provided, the script finds the IP from the Internet.

Author: @vienlt  do you think it is a good idea to tell the user we will find the IP for them and that IP is not workable in many cases ?

Author: @shannona  Can you give us some opinion?

Reviewer 2: @pieceofr : I just change the format of the instructions without changing any content. What is the concern here?

Author: @vienlt  Got it. Now No concern

    + follow script instruction and select "1)Installation"

> 

Reviewer 2: `follow script instruction` to `Follow the script instructions`

Author: ok

    
+  Windows users
    +  ensure that Hyper-V is turned on
    +  [download script for windows](https://s3-ap-northeast-1.amazonaws.com/bitmark-node-docker-scripts/bitmarkNode-HyperV.bat)
    + execute bitmarkNode-HyperV.bat [Your Public IP]. When run the script without providing [Your Public IP], the script find your IP from Internet

        ``` bitmarkNode-HyperV.bat 117.166.111.123```

>     + follow script instruction and select "1)Installation"

Reviewer 2: Same comments as in `Linux and Mac users` section.

Author: ok

+ execute **docker ps** in your command-line/shell to check if bitmark-node is running
![docker ps result](https://i.imgur.com/l3dF4Hl.jpg)

## Management Panel

### Step 1: Open A Node-Management Webpage 

> + view our node-mangement webpage on http://127.0.0.1:9980 in a browser

Reviewer 2: Suggestion:
From:  `view our node-mangement webpage on http://127.0.0.1:9980 in a browser` 
To: `Access the node management webpage at http://127.0.0.1:9980`

Author: ok

+ input your BTC and LTC address for rewarding in a pop-up screen
![Input your BTC&LTC Address](https://i.imgur.com/IRTlyjY.jpg?1)

### Step 2: Run bitmarkd and recorderd
+ startup screen
    + run bitmarkd by click "start button" of "Bitmark Node (bitmarkd)"
    + after bitmarkd is running, run a recorderd by click "start button" of "Recorder Node (recorderd)"
    +  if you don't want to use your machine to help us on mining, you can stop recorderd.

![startup screen](https://i.imgur.com/aeONALb.jpg)

+ Running screen

>     + wait until 5 connections to establish a distribute network

Reviewer 2: Do you mean `Wait until 5 nodes connected to establish a distributed network`?

Author: will change to Linzy's suggestion

>     + wait until 5 connections to establish a distribute network

Reviewer: ```suggestion
    + wait for 5 connections to join the network
```

Author: ok

>     + "Current Block" shows progress of receiving blocks

Reviewer 2: What do you think about the following text?
`The "Current Block" shows the progress of blocks synchronization of the node.`

Author: good! will change to it
